src/solvers/dg_triangles/dg.jl

```diff
@@ -0,0 +1,287 @@
+# !!! warning "Experimental features"
+
+# todo: replace mul! by Octavian.matmul!
+# todo: add @threaded 
+
+# out <- A*x
+mul_by!(A) = let A = A 
+  @inline (out, x)->mul!(out, A, x) 
+end
+
+# out <- out + A * x 
+mul_by_accum!(A) = let A = A 
+  @inline (out, x)->mul!(out, A, x, one(eltype(out)), one(eltype(out))) 
+end
+
+#  out <- out + α * A * x 
+mul_by_accum!(A, α) = let A = A 
+  @inline (out, x)->mul!(out, A, x, α, one(eltype(out))) 
+end
+
+const DGWeakForm{Dims, ElemType} = DG{<:RefElemData{Dims, ElemType}, Mortar, 
+                    <:SurfaceIntegralWeakForm,
+                    <:VolumeIntegralWeakForm} where {Mortar}
+
+# this is necessary for pretty printing
+Base.real(rd::RefElemData{Dims, Elem, ApproxType, Nfaces, RealT}) where {Dims, Elem, ApproxType, Nfaces, RealT} = RealT
+
+function Trixi.ndofs(mesh::AbstractMeshData, dg::DG{<:RefElemData}, cache)
+  rd = dg.basis
+  return rd.Np * mesh.md.num_elements
+end
+Trixi.wrap_array(u_ode::StructArray, semi::Trixi.AbstractSemidiscretization) = u_ode
+Trixi.wrap_array(u_ode::StructArray, mesh::AbstractMeshData, equations, dg::DG{<:RefElemData}, cache) = u_ode
```

Comment:
```suggestion
wrap_array(u_ode::StructArray, semi::AbstractSemidiscretization) = u_ode
wrap_array(u_ode::StructArray, mesh::AbstractMeshData, equations, dg::DG{<:RefElemData}, cache) = u_ode
```
Is the firth on really necessary? It should be redirected to the second method, I think?

Reply:
Removed the first one; seems OK.